### ISSUE TITLE
Undoing damage not always working when using third party to manage saving throws

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#636]
In GitLab by @robnmeyer on Jan 5, 2022, 04:25

Summary:  When using an area of effect where multiple targets are affected and using either monk's tokenbar or LMRTFY, 'undoing' damage often leaves a number of the effected tokens in a corrupt state that requires a refresh to fix.

Reproduction Steps:
1) Configure Midi to use either Monks Token bar or LMRTFY to manage the saving throw dialogue.  I configured a fresh world with only dependencies.
2) Cast fireball on a number of goblins (naturally) - I used 7
3) Regardless of whether they save or not, usually they are all reduced to 0 hp.
4) Either individually, or using the "All" button on the damage manager, undo damage for all the goblins.
5) Almost always there is one, but there have been up to 4, where the goblin HPs remain at zero.
6) Effected goblins cannot be updated to their original hp by any means that I could find (including on the token sheet).
7) A simple refresh fixes the issue.  There are no errors in the console.
8) Using Midi's "Auto" for saving throw management works as expected.

This could be an issue with both Monk's and LMRTFY if they use the similar underlying code.

[fvtt-midi-qol-settings.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/c8f615e495f995fbb1529f720313b9cb76701e1bd0a8e181b97fe47d61031490/fvtt-midi-qol-settings.json)

**Comment by tposney** [comment]
I don't think it's a lmrtfy/monks token bar issue, and I can reproduce the issue when using automatic rolls. 

It's almost certainly a "race condition" between the the actor hp update and the token attribute bar update. I noticed that if I use the undo all button and then do a refresh everything is as it should be.

More importantly, like every race condition issue I've looked at so far, the problem does not exist in v9. With that having been released as stable i think I'm going to leave this in the too hard for me to understand bucket and advise moving to v9.
